packages/xchain-utxo/src/validators.ts

```
@@ -0,0 +1,394 @@
+
+    // Warn for large transactions
+    if (estimatedSize > maxSize * 0.8) {
+      console.warn(`Transaction size ${estimatedSize} bytes is approaching limit ${maxSize} bytes`)
+    }
+  }
+
+  // Private helper methods
+
+  private static containsControlCharacters(str: string): boolean {
+    // Check for control characters that might cause issues
+    return /[\x00-\x08\x0B\x0C\x0E-\x1F\x7F]/.test(str)
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)